src/panels/config/devices/device-detail/ha-device-entities-card.ts

```diff
@@ -125,6 +141,51 @@ export class HaDeviceEntitiesCard extends LitElement {
     fireEvent(this, "hass-more-info", { entityId: entry.entity_id });
   }
 
+  private async _addToLovelaceView(): Promise<void> {
```

Comment:
Feels weird to have this logic be in this compnent. I would prefer if there is a new "Add to Lovelace" dialog added in which we can pass an array of card configs and it will allow the user to pick a view to add them to. And we can render them too, so user knows what they are adding.

Reply:
Using the generated lovelace function? I think that would be really cool, but you should be able to overrule that and choose your own card. I agree this logic can be moved elsewhere.

Reply:
(btw, now we show a dialog to choose a view, and then a dialog to choose the card type and then the default lovelace editor with preview before adding anything :-)

Reply:
oh I didn't realize that last part. That is cool.